### ISSUE TITLE
assert_quantity_allclose breaks if you pass a constant

### ISSUE THREAD

**Issue by Cadair** [astropy/astropy#3538]
```python
assert_quantity_allclose(astropy.constants.c.to(u.m/u.s), astropy.constants.c)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-6-fda44ae261fc> in <module>()
----> 1 assert_quantity_allclose(astropy.constants.c.to(u.m/u.s), astropy.constants.c)

/usr/lib/python2.7/site-packages/astropy/tests/helper.pyc in assert_quantity_allclose(actual, desired, rtol, atol, err_msg, verbose)
    629     actual = u.Quantity(actual, subok=True, copy=False)
    630 
--> 631     desired = u.Quantity(desired, subok=True, copy=False)
    632     try:
    633         desired = desired.to(actual.unit)

TypeError: __init__() got an unexpected keyword argument 'subok'
```

ping @astrofrog

**Comment by mhvk** [comment]
hmm, a more general problem with constants:
```
import astropy.units as u, numpy as np, astropy.constants as c
u.Quantity(c.c)
# <repr(<astropy.units.quantity.Quantity at 0x7f7d78352588>) failed: AttrbuteError: 'Quantity' object has no '_unitstr' member>
```
But if one gives a `unit`, it is OK:
```
u.Quantity(c.c, 'm/s', subok=True)
# <Quantity 299792458.0 m / s>
```
Investigating further...

**Comment by Cadair** [comment]
@mhvk see #3537

**Comment by mhvk** [comment]
Fix in #3537.

**Comment by embray** [comment]
Fixed by #3537.
